src/document.c

```diff
@@ -628,7 +628,9 @@ static GeanyDocument *document_create(const gchar *utf8_filename)
 	{
 		doc = document_get_current();
 		/* remove the empty document first */
-		if (doc != NULL && doc->file_name == NULL && ! doc->changed)
+		if (doc != NULL && doc->file_name == NULL && ! doc->changed &&
+			// check save as dialog isn't running
+			gtk_window_is_active(GTK_WINDOW(main_widgets.window)))
```

Comment:
Unfortunately this is false when the window doesn't have focus, so it won't close the empty document any more when using a terminal window to open a file.

Reply:
So it looks like this function would be a solution (comparing the result to the main window):
https://docs.gtk.org/gtk3/method.Application.get_active_window.html

But we don't use GtkApplication, also I'm not sure if it would mean having to update all our window creation code too.

Reply:
Not sure if that would actually account for dialogs as well, I would think it's mostly to select the MRU one in a multi-window app.